### PR TITLE
[React useMemo] Add useMemo hook for syntaxes

### PR DESCRIPTION
**Pull request by lotrien**

### CODE REVIEW

File: src/components/NewSnippet.jsx [hunk 1]
```diff
@@ -70,14 +70,14 @@ const NewSnippet = ({ dispatch, history, syntaxes }) => {
   const handleSyntax = syntax => ({ syntax })
   const handleContent = content => ({ content })
 
-  const getSyntaxes = () => {
+  const memoizedSyntaxes = useMemo(() => {
```
Comment (ikalnytskyi) [inline src/components/NewSnippet.jsx:73]: What's the benefit of using `useMemo()`?

File: src/hooks/useForm.js [hunk 2]
```diff
@@ -15,21 +15,21 @@ const useForm = (cb, validate) => {
     }
   }, [error, isSubmitting])
 
-  const handleSubmit = useCallback(e => {
```
Comment (ikalnytskyi) [inline src/hooks/useForm.js:18]: Why `useCallback` has been removed?

**Review by ikalnytskyi — APPROVED** [review]

File: src/components/NewSnippet.jsx (hunk 1 above)
Comment (lotrien) [reply src/components/NewSnippet.jsx:73]: `useMemo` will memorize result of passed function and will be re-rendered only if `syntaxes` has changed.

File: src/hooks/useForm.js (hunk 2 above)
Comment (lotrien) [reply src/hooks/useForm.js:18]: in our case without second parameter this hook does not have sense.

**Pull request merged**